packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -0,0 +1,233 @@
+            navigationItem.children &&
+            navigationItem.children.some(
+              (nestedNavigationItem) =>
+                (!nestedNavigationItem.kind || nestedNavigationItem.kind === 'page') &&
+                nestedNavigationItem.path === window.location.pathname,
+            ),
+        )
+        .map(
+          ({ navigationItem, originalIndex }) =>
+            `${(navigationItem as NavigationPageItem).title}-${depth}-${originalIndex}`,
+        );
+
+      setExpandedSidebarItemIds(initialExpandedSidebarItemIds);
+    }
+  }, [depth, subNavigation]);
+
+  const handleSidebarItemClick = React.useCallback(
+    (itemId: string) => () => {
+      setExpandedSidebarItemIds((previousValue) =>
+        previousValue.includes(itemId)
+          ? previousValue.filter((previousValueItemId) => previousValueItemId !== itemId)
+          : [...previousValue, itemId],
+      );
+    },
+    [],
+  );
+
+  return (
+    <List sx={{ mb: depth === 0 ? 4 : 1, pl: 2 * depth }}>
+      {subNavigation.map((navigationItem, navigationItemIndex) => {
+        if (navigationItem.kind === 'header') {
+          return (
+            <ListSubheader key={`subheader-${depth}-${navigationItemIndex}`} component="div">
+              {navigationItem.title}
+            </ListSubheader>
+          );
+        }
+
+        if (navigationItem.kind === 'divider') {
+          const nextItem = subNavigation[navigationItemIndex + 1];
+
+          return (
+            <Divider
+              key={`divider-${depth}-${navigationItemIndex}`}
+              sx={{ mt: 1, mb: nextItem.kind === 'header' ? 0 : 1 }}
+            />
+          );
+        }
+
+        const navigationItemId = `${navigationItem.title}-${depth}-${navigationItemIndex}`;
+
+        const isNestedNavigationExpanded = expandedSidebarItemIds.includes(navigationItemId);
+
+        const nestedNavigationCollapseIcon = isNestedNavigationExpanded ? (
+          <ExpandLessIcon />
+        ) : (
+          <ExpandMoreIcon />
+        );
+
+        const listItem = (
+          <ListItem>
+            <ListItemButton
+              selected={IS_CLIENT && hasMounted && window.location.pathname === navigationItem.path}
+              onClick={handleSidebarItemClick(navigationItemId)}
+            >
+              <ListItemIcon>{navigationItem.icon}</ListItemIcon>
+              <ListItemText primary={navigationItem.title} />
+              {navigationItem.children ? nestedNavigationCollapseIcon : null}
+            </ListItemButton>
+          </ListItem>
+        );
+
+        return (
+          <React.Fragment key={navigationItemId}>
+            {navigationItem.path ? (
+              <a href={navigationItem.path} style={{ color: 'inherit', textDecoration: 'none' }}>
+                {listItem}
+              </a>
+            ) : (
+              listItem
+            )}
+            {IS_CLIENT && hasMounted && navigationItem.children ? (
```

Comment:
`IS_CLIENT` can never be `false` if `hasMounted` is `true`.


```suggestion
            {hasMounted && navigationItem.children ? (
```